src/encoder/encoderfdkaac.cpp

```diff
@@ -0,0 +1,408 @@
+// encoderfdkaac.cpp
+// Created on Aug 15 2017 by Palakis
+
+#include <QDesktopServices>
+#include <QDir>
+#include <QString>
+#include <QStringList>
+
+#include "recording/defs_recording.h"
+#include "util/logger.h"
+#include "util/sample.h"
+#include "engine/sidechain/enginesidechain.h"
+
+#include "encoder/encoderfdkaac.h"
+
+namespace {
+// recommended in encoder documentation, section 2.4.1
+const int kOutBufferBits = 6144;
+const mixxx::Logger kLogger("EncoderFdkAac");
+}
+
+EncoderFdkAac::EncoderFdkAac(EncoderCallback* pCallback, QString pFormat)
+    : aacEncOpen(nullptr),
+      aacEncClose(nullptr),
+      aacEncEncode(nullptr),
+      aacEncInfo(nullptr),
+      aacEncoder_SetParam(nullptr),
+      m_aacAot(0),
+      m_bitrate(0),
+      m_channels(0),
+      m_samplerate(0),
+      m_pCallback(pCallback),
+      m_library(nullptr),
+      m_pInputFifo(nullptr),
+      m_pFifoChunkBuffer(nullptr),
+      m_readRequired(0),
+      m_aacEnc(),
+      m_pAacDataBuffer(nullptr),
+      m_aacInfo() {
+
+    if (pFormat == ENCODING_AAC) {
+        // MPEG-4 AAC-LC
+        m_aacAot = AOT_AAC_LC;
+    }
+    else if (pFormat == ENCODING_HEAAC) {
+        // MPEG-4 HE-AAC
+        m_aacAot = AOT_SBR;
+    }
+    else if (pFormat == ENCODING_HEAACV2) {
+        // MPEG-4 HE-AACv2
+        m_aacAot = AOT_PS;
+    }
+    else {
+        // Fallback to AAC-LC in case
+        // of unknown value
+        m_aacAot = AOT_AAC_LC;
+    }
+
+    // Load shared library
+    // Code import from encodermp3.cpp
+    QStringList libnames;
+    QString libname = "";
+#ifdef __LINUX__
+    libnames << "fdk-aac";
+#elif __WINDOWS__
+    // Give top priority to libfdk-aac copied
+    // into Mixxx's installation folder
+    libnames << "libfdk-aac-1.dll";
+
+    // Fallback and user-friendly method: use libfdk-aac
+    // provided with B.U.T.T installed in
+    // a standard location
+    QString buttFdkAacPath = buttWindowsFdkAac();
+    if(!buttFdkAacPath.isNull()) {
+        kLogger.debug() << "Found libfdk-aac at" << buttFdkAacPath;
+        libnames << buttFdkAacPath;
+    }
+
+    // Last resort choices: try versions with unusual names
+    libnames << "libfdk-aac.dll";
+    libnames << "libfdkaac.dll";
+#elif __APPLE__
```

Comment:
Please add `libfdk-aac` here too so that if it's present in the search paths it will work.